exonum-light-client/src/main/java/com/exonum/client/ExonumClient.java

```diff
@@ -77,6 +80,95 @@
+   * of blocks plus one.
+   *
+   * <p>For example, the "genesis" block has height {@code h = 0}. The latest committed block
+   * has height {@code h = getBlockHashes().size() - 1}.
+   *
+   * @throws RuntimeException if the client is unable to complete a request
+   *        (e.g., in case of connectivity problems)
+   */
+  long getBlockchainHeight();
+
+  /**
+   * Returns the information about the block with transaction hashes included at this block.
+   * @param height blockchain height starting from 0 (genesis block)
+   * @return block information response
+   * @throws RuntimeException if the client is unable to complete a request
+   *        (e.g., in case of connectivity problems)
+   * @throws RuntimeException if block is not found by the requested height,
+   *        i.e. the requested height is greater then actual blockchain height
+   * @throws IllegalArgumentException if the given height is negative
+   */
+  BlockResponse getBlockByHeight(long height);
+
+  /**
+   * Returns blockchain blocks information for the requested range.
+   * @param count Number of blocks to return.
+   *        It should not be negative and should not be greater then
+   *        It should be in range [1, {@linkplain ExonumApi#MAX_BLOCKS_PER_REQUEST}]
+   * @param skipEmpty if {@code true}, then only non-empty blocks will be returned
+   * @param heightMax maximum height of the returned blocks. The blocks are returned
+   *        in reverse order, starting from the <b>heightMax</b> and
+   *        at least up to the <b>heightMax - count + 1</b>.
+   *        If the <b>heightMax</b> is greater than actual blockchain height then
+   *        the actual height will be used
+   * @param withTime indicates for filling {@linkplain Block#getCommitTime() block commit time}
+   *        if {@code true}; or left it {@linkplain Optional#empty() empty} if {@code false}.
+   *        The time value corresponds to the average time of submission of precommits by the
+   *        validators for every returned block
+   * @return blocks information response
+   * @throws RuntimeException if the client is unable to complete a request
+   *        (e.g., in case of connectivity problems)
+   * @throws IllegalArgumentException if count is out of range
+   *        [1, {@linkplain ExonumApi#MAX_BLOCKS_PER_REQUEST}]
+   */
+  BlocksResponse getBlocks(int count, boolean skipEmpty, long heightMax, boolean withTime);
+
+  /**
+   * Returns blockchain blocks information starting from the last block in the blockchain.
+   * @param count Number of blocks to return.
+   *        It should be in range [1, {@linkplain ExonumApi#MAX_BLOCKS_PER_REQUEST}]
+   * @param skipEmpty if {@code true}, then only non-empty blocks are returned
+   * @param withTime if {@code true}, then includes block commit times in the response;
+   *        or an empty times {@code false}. See {@linkplain Block#getCommitTime()}.
```

Comment:
Here and elsewhere, "or an empty times {@code false}" seems off. @Maria-Nosyk WDYT?